libplatsupport/mach_include/spacemit/platsupport/mach/timer.h

```diff
@@ -25,50 +33,23 @@
 
 #define SPACEMIT_NUM_TIMERS 3
 #define SPACEMIT_TIMER_MAX_TICKS UINT32_MAX
-
-#define APB_CLK_BASE                0xD4015000
 #define APBC_TIMERx_CLK_RST_OFFSET  0x34u
 
-#define SPACEMIT_TIMERS_TCER_OFFSET          0x0   /* Timer Count Enable Register (TCER)   */
-#define SPACEMIT_TIMERS_TCMR_OFFSET          0x4   /* Timer Count Mode Register (TCMR)     */
-#define SPACEMIT_TIMERS_TCRR_OFFSET          0x8   /* Timer Count Restart Register (TCRR)  */
-#define SPACEMIT_TIMERS_TCCR_OFFSET          0xC   /* Timer Clock Control Register (TCCR)  */
-
-/* Timer Match Register */
-#define SPACEMIT_TIMERS_TMR_OFFSET  0x10u
-#define K1_TMR_OFS_CAL(n, m)  (SPACEMIT_TIMERS_TMR_OFFSET + ((uint32_t)(n) * 0x10u) + ((uint32_t)(m) * 0x4u))
-
-/* Timer Preload Value Register */
-#define SPACEMIT_TIMERS_TPLVR_OFFSET       0x40u
-#define K1_TPLVR_OFS_CAL(n)  (SPACEMIT_TIMERS_TPLVR_OFFSET + ((uint32_t)(n) * 0x4u))
-
-/* Timer Preload Control Register */
-#define SPACEMIT_TIMERS_TPLCR_OFFSET       0x50u
-#define K1_TPLCR_OFS_CAL(n)  (SPACEMIT_TIMERS_TPLCR_OFFSET + ((uint32_t)(n) * 0x4u))
-#define K1_TPLCR_ENCODE(mcs, crpd) \
-    ( ((uint32_t)(mcs) & 0x3u) | ((uint32_t)(crpd) << 2) )
-
-/* Timer Interrupt Enable Register */
-#define SPACEMIT_TIMERS_IER_OFFSET        0x60u
-#define K1_TIER_OFS_CAL(n)  (SPACEMIT_TIMERS_IER_OFFSET + ((uint32_t)(n) * 0x4u))
-
-/* Timer Interrupt Clear Register */
-#define SPACEMIT_TIMERS_ICR_OFFSET        0x70u
-#define K1_TICLR_OFS_CAL(n)  (SPACEMIT_TIMERS_ICR_OFFSET + ((uint32_t)(n) * 0x4u))
+typedef struct {
+    uint32_t tcer;         // Timer Count Enable Register
+    uint32_t tcmr;         // Timer Count Mode Register
+    uint32_t tcrr;         // Timer Count Restart Register
+    uint32_t tccr;         // Timer Clock Control Register
 
-/* Timer Status Register */
-#define SPACEMIT_TIMERS_ISR_OFFSET        0x80u
-#define K1_TSR_OFS_CAL(n)  (SPACEMIT_TIMERS_ISR_OFFSET + ((uint32_t)(n) * 0x4u))
+    uint32_t tmr[3][4];    // Timer Match Register (TMRx)
+    uint32_t tplvr[4];     // Timer Preload Value Register (TPLVRx)
+    uint32_t tplcr[4];     // Timer Preload Control Register (TPLCRx)
 
-/* Timer Count Register */
-#define SPACEMIT_TIMERS_CNT_OFFSET        0x90u
-#define K1_TCNT_OFS_CAL(n)  (SPACEMIT_TIMERS_CNT_OFFSET + ((uint32_t)(n) * 0x4u))
+    uint32_t tier[4];       // Timer Interrupt Enable Register (TIERx)
+    uint32_t ticr[4];       // Timer Interrupt Clear Register (TICLRx)
+    uint32_t tsr[4];        // Timer Status Register (TSRx)
 
-typedef struct {
-    uint32_t tmr_cnt_en;
-    uint32_t tmr_mode;
-    uint32_t tmr_restart;
-    uint32_t tmr_clock_ctrl;
+    uint32_t tcnt[4];       // Timer Count Register (TCRx)
 } spacemit_timer_regs_t;
```

Comment:
It's good to add a static _assert to check that the last field has the correct offset, to catch most errors elsewhere.